### ISSUE TITLE
0.2.8 broke running at pre-commit.ci

### ISSUE THREAD

**Issue by nijel** [rvben/rumdl#663]
### rumdl version

0.2.8-0.2.12

### Installation method

Pre-built binary

### Bug description

We're using rumdl via the pre-commit.ci service, and since 0.2.8 the configuration present in pyproject.toml is no longer applied.

It turns out that the service sets $HOME to the Git checkout directory, which makes the top-level `pyproject.toml` not discoverable.

### Steps to reproduce

```sh
mkdir test
cd test
cat > pyproject.toml << EOT
[tool.rumdl]
disable = ["first-line-heading"]
EOT
echo Test > TEST.md
echo "HOME=$HOME"
uvx --from rumdl==0.2.12 rumdl check
echo "HOME=$PWD"
HOME=$PWD uvx --from rumdl==0.2.12 rumdl check
```

```console
HOME=/home/nijel

Success: No issues found in 1 file (1ms)
HOME=/tmp/test/test
Installed 1 package in 7ms
TEST.md:1:1: [MD041] First line in file should be a level 1 heading

Issues: Found 1 issues in 1 file (1ms)
```

### Expected behavior

The `pyproject.toml` would still be loaded.

### Actual behavior

`pyproject.toml` is ignored if the current directory is $HOME.

### Configuration

```toml

```

### Operating system

Linux

### Additional context

0cdf3b72a6d2f6bababe4ec6f6605d183848f1ad is the commit that changed this behavior.

**Comment by rvben** [comment]
Thanks for the precise reproduction and the bisect @nijel. This is fixed in v0.2.13, released earlier today. The home-directory boundary added in 0.2.8 (meant to keep `~/.rumdl.toml` user-level rather than shadowing the platform config dir) wrongly skipped the starting directory itself, so a project config in $HOME was never probed. Discovery now always probes the working directory, even when it is $HOME. Verified with your exact script: 0.2.12 reports MD041, 0.2.13 applies the `pyproject.toml` config. Upgrading to 0.2.13 should unblock pre-commit.ci.
